### PR TITLE
fix: should not list empty workspace folders

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: src/testTree.ts [hunk 1]
```diff
@@ -99,15 +89,20 @@ export class TestTree extends DisposableBase {
...
+      for (const model of this._models.enabledModels().filter(m => m.config.workspaceFolder === workspaceFolder.uri.fsPath)) {
         for (const project of model.enabledProjects())
           rootSuite.suites.push(project.suite as TeleSuite);
       }
-      const upstreamTree = new upstream.TestTree(workspaceFolder, rootSuite, [], undefined, path.sep);
+      const upstreamTree = new upstream.TestTree(workspaceFolder.uri.fsPath, rootSuite, [], undefined, path.sep);
       upstreamTree.sortAndPropagateStatus();
       upstreamTree.flattenForSingleProject();
+      if (upstreamTree.rootItem.children.length === 0) {
+        this._testController.items.delete(this._idWithGeneration(workspaceFolder.uri.fsPath));
```
Comment (pavelfeldman) [inline src/testTree.ts:102]: Does delete work fine if the element is not there?

File: src/testTree.ts [hunk 2]
```diff
@@ -99,15 +89,20 @@ export class TestTree extends DisposableBase {
   }
 
   private _update() {
-    for (const [workspaceFolder, workspaceRootItem] of this._rootItems) {
+    for (const workspaceFolder of this._vscode.workspace.workspaceFolders ?? []) {
```
Comment (pavelfeldman) [inline src/testTree.ts:92]: I think the folder might be stuck if user removes a workspaceFolder.

File: src/testTree.ts (hunk 1 above)
Comment (mxschmitt) [reply src/testTree.ts]: Yup!

File: src/testTree.ts (hunk 2 above)
Comment (mxschmitt) [reply src/testTree.ts:92]: Addressed.

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**